qemu/tests/hugepage_reset.py

```diff
@@ -21,6 +21,33 @@ def run(test, params, env):
     :param env: Dictionary with the test environment.
     """
 
+    def allocate_largepages_per_node():
+        """
+        The functions is inteded to set 1G hugepages per NUMA
+        node when the system has four or more of these nodes.
+        For this function to work the hugepage size should be 1G.
+        This way a QEMU failure could be avoided if it's unable
+        to allocate memory.
+        """
+        node_list = host_numa_node.online_nodes_withcpumem
+        if len(node_list) >= 4:
+            allocated_memory = False
+            try:
+                for node in node_list:
+                    node_mem_free = int(
+                        host_numa_node.read_from_node_meminfo(node, "MemFree")
+                    )
+                    if node_mem_free > (mem * 1024):
+                        hp_config.set_node_num_huge_pages(4, node, "1048576")
+                        allocated_memory = True
+                        break
+            except ValueError as e:
+                test.cancel(e)
+            if not allocated_memory:
+                test.fail(
```

Comment:
This is quite strange to me. When ValueError occurs, you cancel the case, but the case fails when all nodes do not have enough memory. Can you explain this decision?

Reply:
Sorry for that, after giving another thought it is duplicated code from: https://github.com/autotest/tp-qemu/blob/157fd7c78b8862dde41a31f02f193b6de5d170e3/qemu/tests/hugepage_reset.py#L87-L108

Which in fact is not totally correct because `node_mdem_free` and `mem` variables are not in the same unit.